src/module.c

```diff
@@ -12856,37 +12856,59 @@ long long getModuleNumericConfig(ModuleConfig *module_config) {
     return module_config->get_fn.get_numeric(rname, module_config->privdata);
 }
 
+int loadModuleSingleConfig(dictEntry *config_queue_entry, ModuleConfig *module_config, bool set_default_if_missing) {
+    const char *err = NULL;
+    if (config_queue_entry) {
+        if (!performModuleConfigSetFromName(dictGetKey(config_queue_entry), dictGetVal(config_queue_entry), &err)) {
+            serverLog(LL_WARNING, "Issue during loading of configuration %s : %s", (sds) dictGetKey(config_queue_entry), err);
+            dictFreeUnlinkedEntry(server.module_configs_queue, config_queue_entry);
+            dictEmpty(server.module_configs_queue, NULL);
+            return REDISMODULE_ERR;
+        }
+        dictFreeUnlinkedEntry(server.module_configs_queue, config_queue_entry);
+    } else if (set_default_if_missing) {
+        if (!performModuleConfigSetDefaultFromName(module_config->name, &err)) {
+            serverLog(LL_WARNING, "Issue attempting to set default value of configuration %s : %s", module_config->name, err);
+            dictEmpty(server.module_configs_queue, NULL);
```

Comment:
when we get here from RM_LoadDefaultConfigs, i'm not sure we should call that dictEmpty (it has nothing to do with what we're doing.